{
    "url": "https://api.github.com/repos/pallets/flask/issues/797",
    "repository_url": "https://api.github.com/repos/pallets/flask",
    "labels_url": "https://api.github.com/repos/pallets/flask/issues/797/labels{/name}",
    "comments_url": "https://api.github.com/repos/pallets/flask/issues/797/comments",
    "events_url": "https://api.github.com/repos/pallets/flask/issues/797/events",
    "html_url": "https://github.com/pallets/flask/issues/797",
    "id": 16625918,
    "node_id": "MDU6SXNzdWUxNjYyNTkxOA==",
    "number": 797,
    "title": "Can't find requet.form or requesrt.values items with post request for flask 0.10.1",
    "user": {
        "login": "tbicr",
        "id": 416731,
        "node_id": "MDQ6VXNlcjQxNjczMQ==",
        "avatar_url": "https://avatars1.githubusercontent.com/u/416731?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/tbicr",
        "html_url": "https://github.com/tbicr",
        "followers_url": "https://api.github.com/users/tbicr/followers",
        "following_url": "https://api.github.com/users/tbicr/following{/other_user}",
        "gists_url": "https://api.github.com/users/tbicr/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/tbicr/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/tbicr/subscriptions",
        "organizations_url": "https://api.github.com/users/tbicr/orgs",
        "repos_url": "https://api.github.com/users/tbicr/repos",
        "events_url": "https://api.github.com/users/tbicr/events{/privacy}",
        "received_events_url": "https://api.github.com/users/tbicr/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [],
    "state": "closed",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 10,
    "created_at": "2013-07-11T11:25:43Z",
    "updated_at": "2013-07-26T16:29:59Z",
    "closed_at": "2013-07-24T15:57:01Z",
    "author_association": "NONE",
    "body": "I have next example which work fine with `flask==0.9`, `werkzeug==0.8.3` and fail with `flask==0.10.1`, `werkzeug==0.9.1`:\n\n```\nfrom flask import Flask, request\n\n\ndef create_app():\n    class Config(object):\n        SECRET_KEY = 'test'\n\n    app = Flask(__name__)\n    app.config.from_object(Config)\n\n    @app.route('/', methods=['POST'])\n    def home():\n        assert request.values['PHONE'] == '+7(910)123-45-67'\n        return request.values['PHONE']\n\n    return app\n\n\nif __name__ == '__main__':\n    app = create_app()\n    #app.run('0.0.0.0', 8080, True)\n    client = app.test_client()\n    response = client.post('/', data={'PHONE': '+7(910)123-45-67'})\n    assert response.data == '+7(910)123-45-67', response.data\n```\n\nWhen I try `wget --post-data='PHONE=%2B7%28910%29123-45-67' localhost:8080` it also will work with `flask==0.9` and don't with `flask==0.10.1`.\n\nFor `flask==0.10.1` I have `request.data=='PHONE=%2B7%28910%29123-45-67'`, but empty `request.values` and `request.form`.\n",
    "comments_inline": [
        {
            "url": "https://api.github.com/repos/pallets/flask/issues/comments/20826703",
            "html_url": "https://github.com/pallets/flask/issues/797#issuecomment-20826703",
            "issue_url": "https://api.github.com/repos/pallets/flask/issues/797",
            "id": 20826703,
            "node_id": "MDEyOklzc3VlQ29tbWVudDIwODI2NzAz",
            "user": {
                "login": "ghost",
                "id": 10137,
                "node_id": "MDQ6VXNlcjEwMTM3",
                "avatar_url": "https://avatars3.githubusercontent.com/u/10137?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/ghost",
                "html_url": "https://github.com/ghost",
                "followers_url": "https://api.github.com/users/ghost/followers",
                "following_url": "https://api.github.com/users/ghost/following{/other_user}",
                "gists_url": "https://api.github.com/users/ghost/gists{/gist_id}",
                "starred_url": "https://api.github.com/users/ghost/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/ghost/subscriptions",
                "organizations_url": "https://api.github.com/users/ghost/orgs",
                "repos_url": "https://api.github.com/users/ghost/repos",
                "events_url": "https://api.github.com/users/ghost/events{/privacy}",
                "received_events_url": "https://api.github.com/users/ghost/received_events",
                "type": "User",
                "site_admin": false
            },
            "created_at": "2013-07-11T17:05:45Z",
            "updated_at": "2013-07-11T17:05:45Z",
            "author_association": "NONE",
            "body": "Can't reproduce:\n\n> When I try wget --post-data='PHONE=%2B7%28910%29123-45-67' localhost:8080 it also will work with flask==0.9 and **don't with flask==0.10.1**.\n\nTry with `curl -d 'PHONE=%2B7%28910%29123-45-67' localhost:8080`. It works!\n\n> For flask==0.10.1 I have request.data=='PHONE=%2B7%28910%29123-45-67', but _empty request.values and request.form_.\n\nAre you sure? On Python 2.7.3/Fedora 18 box `request.values` is `CombinedMultiDict([ImmutableMultiDict([]), ImmutableMultiDict([('PHONE', u'+7(910)123-45-67')])])`\n"
        },
        {
            "url": "https://api.github.com/repos/pallets/flask/issues/comments/20837862",
            "html_url": "https://github.com/pallets/flask/issues/797#issuecomment-20837862",
            "issue_url": "https://api.github.com/repos/pallets/flask/issues/797",
            "id": 20837862,
            "node_id": "MDEyOklzc3VlQ29tbWVudDIwODM3ODYy",
            "user": {
                "login": "tbicr",
                "id": 416731,
                "node_id": "MDQ6VXNlcjQxNjczMQ==",
                "avatar_url": "https://avatars1.githubusercontent.com/u/416731?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/tbicr",
                "html_url": "https://github.com/tbicr",
                "followers_url": "https://api.github.com/users/tbicr/followers",
                "following_url": "https://api.github.com/users/tbicr/following{/other_user}",
                "gists_url": "https://api.github.com/users/tbicr/gists{/gist_id}",
                "starred_url": "https://api.github.com/users/tbicr/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/tbicr/subscriptions",
                "organizations_url": "https://api.github.com/users/tbicr/orgs",
                "repos_url": "https://api.github.com/users/tbicr/repos",
                "events_url": "https://api.github.com/users/tbicr/events{/privacy}",
                "received_events_url": "https://api.github.com/users/tbicr/received_events",
                "type": "User",
                "site_admin": false
            },
            "created_at": "2013-07-11T19:57:22Z",
            "updated_at": "2013-07-12T08:12:44Z",
            "author_association": "NONE",
            "body": "Look like problem with my pycharm when I trying debugging and add `request` to watches (then  `request.values` in wathces is `CombinedMultiDict([ImmutableMultiDict([]), ImmutableMultiDict([])])` but `request.data` still `'PHONE=+7%28910%29123-45-67'`!!!). Without debugging curl and wget return right value.\n\nWhen I try:\n\n```\nclient = app.test_client()\nresponse = client.post('/', data={'PHONE': '+7(910)123-45-67'})\n```\n\n(same code I use for unit tests) then have next error:\n\n```\nTraceback (most recent call last):\n  File \"../request-test/request-test.py\", line 24, in <module>\n    response = client.post('/', data={'PHONE': '+7(910)123-45-67'})\n  File \"/home/tbicr/Projects/env/local/lib/python2.7/site-packages/werkzeug/test.py\", line 770, in post\n    return self.open(*args, **kw)\n  File \"/home/tbicr/Projects/env/local/lib/python2.7/site-packages/flask/testing.py\", line 108, in open\n    follow_redirects=follow_redirects)\n  File \"/home/tbicr/Projects/env/local/lib/python2.7/site-packages/werkzeug/test.py\", line 734, in open\n    response = self.run_wsgi_app(environ, buffered=buffered)\n  File \"/home/tbicr/Projects/env/local/lib/python2.7/site-packages/werkzeug/test.py\", line 657, in run_wsgi_app\n    rv = run_wsgi_app(self.application, environ, buffered=buffered)\n  File \"/home/tbicr/Projects/env/local/lib/python2.7/site-packages/werkzeug/test.py\", line 853, in run_wsgi_app\n    app_iter = app(environ, start_response)\n  File \"/home/tbicr/Projects/env/local/lib/python2.7/site-packages/flask/app.py\", line 1836, in __call__\n    return self.wsgi_app(environ, start_response)\n  File \"/home/tbicr/Projects/env/local/lib/python2.7/site-packages/flask/app.py\", line 1820, in wsgi_app\n    response = self.make_response(self.handle_exception(e))\n  File \"/home/tbicr/Projects/env/local/lib/python2.7/site-packages/flask/app.py\", line 1403, in handle_exception\n    reraise(exc_type, exc_value, tb)\n  File \"/home/tbicr/Projects/env/local/lib/python2.7/site-packages/flask/app.py\", line 1817, in wsgi_app\n    response = self.full_dispatch_request()\n  File \"/home/tbicr/Projects/env/local/lib/python2.7/site-packages/flask/app.py\", line 1477, in full_dispatch_request\n    rv = self.handle_user_exception(e)\n  File \"/home/tbicr/Projects/env/local/lib/python2.7/site-packages/flask/app.py\", line 1381, in handle_user_exception\n    reraise(exc_type, exc_value, tb)\n  File \"/home/tbicr/Projects/env/local/lib/python2.7/site-packages/flask/app.py\", line 1475, in full_dispatch_request\n    rv = self.dispatch_request()\n  File \"/home/tbicr/Projects/env/local/lib/python2.7/site-packages/flask/app.py\", line 1461, in dispatch_request\n    return self.view_functions[rule.endpoint](**req.view_args)\n  File \"../request-test/request-test.py\", line 14, in home\n    assert request.values['PHONE'] == '+7(910)123-45-67', request.values['PHONE']\nAssertionError:  7(910)123-45-67\n```\n\nThis is initial problem which I want reproduce at first with small example, but could not at first because have described problem with pycharm. There are diffrenece `+7(910)123-45-67` and `7(910)123-45-67`.\n\nAll this work right with `flask==0.9` and `werkzeug==0.8.3`.\n\nSo there are two issue:\n1. with pycharm debugging and `request` in watches (http://youtrack.jetbrains.com/issue/PY-10275).\n2. with `test_client` and specific chars in request.\n\nEnvirements:\n\n```\nLinux local 3.2.0-4-amd64 #1 SMP Debian 3.2.32-1 x86_64 GNU/Linux\nPython 2.7.5 (default, May 20 2013, 11:51:12) \n[GCC 4.7.3] on linux2\n```\n\nand\n\n```\nLinux local 3.8.0-26-generic #38-Ubuntu SMP Mon Jun 17 21:43:33 UTC 2013 x86_64 x86_64 x86_64 GNU/Linux\nPython 2.7.4 (default, Apr 19 2013, 18:28:01) \n[GCC 4.7.3] on linux2\n```\n"
        },
        {
            "url": "https://api.github.com/repos/pallets/flask/issues/comments/21456637",
            "html_url": "https://github.com/pallets/flask/issues/797#issuecomment-21456637",
            "issue_url": "https://api.github.com/repos/pallets/flask/issues/797",
            "id": 21456637,
            "node_id": "MDEyOklzc3VlQ29tbWVudDIxNDU2NjM3",
            "user": {
                "login": "frankstratton",
                "id": 293509,
                "node_id": "MDQ6VXNlcjI5MzUwOQ==",
                "avatar_url": "https://avatars0.githubusercontent.com/u/293509?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/frankstratton",
                "html_url": "https://github.com/frankstratton",
                "followers_url": "https://api.github.com/users/frankstratton/followers",
                "following_url": "https://api.github.com/users/frankstratton/following{/other_user}",
                "gists_url": "https://api.github.com/users/frankstratton/gists{/gist_id}",
                "starred_url": "https://api.github.com/users/frankstratton/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/frankstratton/subscriptions",
                "organizations_url": "https://api.github.com/users/frankstratton/orgs",
                "repos_url": "https://api.github.com/users/frankstratton/repos",
                "events_url": "https://api.github.com/users/frankstratton/events{/privacy}",
                "received_events_url": "https://api.github.com/users/frankstratton/received_events",
                "type": "User",
                "site_admin": false
            },
            "created_at": "2013-07-24T00:40:06Z",
            "updated_at": "2013-07-24T00:40:06Z",
            "author_association": "NONE",
            "body": "I've found a similar issue with request.data and request.form.\n\nHere's my test code:\n\n```\n@app.route(\"/\", methods=['POST'])\ndef hello():\n    print request.data\n    print request.form\n    return \"Hello World!\"\n```\n\nHere's my commandline test:\n\n```\ncurl -d \"asdf=asdf\" http://localhost:5000\n```\n\nHere's the output using werkzeug 0.8\n\n```\n\nImmutableMultiDict([('asdf', u'asdf')])\n```\n\nHere's the output using werkzeug 0.9\n\n```\nasdf=asdf\nImmutableMultiDict([])\n```\n\nIf you modify the example in to access request.form first before request.data then it works okay in werkzeug 0.9.  \nI've tracked it down to the fact that accessing request.data consumes the input stream leaving an empty data stream by the time you call request.form.  \n\nI'm not sure what changed between 0.8 and 0.9 that creates an order of operations bug.\n"
        },
        {
            "url": "https://api.github.com/repos/pallets/flask/issues/comments/21473270",
            "html_url": "https://github.com/pallets/flask/issues/797#issuecomment-21473270",
            "issue_url": "https://api.github.com/repos/pallets/flask/issues/797",
            "id": 21473270,
            "node_id": "MDEyOklzc3VlQ29tbWVudDIxNDczMjcw",
            "user": {
                "login": "untitaker",
                "id": 837573,
                "node_id": "MDQ6VXNlcjgzNzU3Mw==",
                "avatar_url": "https://avatars0.githubusercontent.com/u/837573?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/untitaker",
                "html_url": "https://github.com/untitaker",
                "followers_url": "https://api.github.com/users/untitaker/followers",
                "following_url": "https://api.github.com/users/untitaker/following{/other_user}",
                "gists_url": "https://api.github.com/users/untitaker/gists{/gist_id}",
                "starred_url": "https://api.github.com/users/untitaker/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/untitaker/subscriptions",
                "organizations_url": "https://api.github.com/users/untitaker/orgs",
                "repos_url": "https://api.github.com/users/untitaker/repos",
                "events_url": "https://api.github.com/users/untitaker/events{/privacy}",
                "received_events_url": "https://api.github.com/users/untitaker/received_events",
                "type": "User",
                "site_admin": false
            },
            "created_at": "2013-07-24T09:28:00Z",
            "updated_at": "2013-07-24T09:31:07Z",
            "author_association": "MEMBER",
            "body": "Introduced somewhere around mitsuhiko/werkzeug@534c7f6e443316b1d5bba8baf700ea62a64c3b31. For the testcase @frankstratton provided request.stream is `LimitedStream`.\n\nI am not sure why these changes were even introduced. So the closest i've come is to write a test case for this bug:\n\n``` python\ndef test_raw_data_shouldnt_exhaust_parsed_form(self):\n    r = wrappers.Request({\n        'REQUEST_METHOD': 'POST',\n        'PATH_INFO': '/',\n        'QUERY_STRING': '',\n        'CONTENT_LENGTH': '9',\n        'wsgi.input': BytesIO(b'asdf=asdf'),\n        'CONTENT_TYPE': 'application/x-www-form-urlencoded'\n    })\n    assert r.data == b'asdf=asdf'\n    assert dict(r.form) == {'asdf': 'asdf'}, dict(r.form)\n```\n"
        },
        {
            "url": "https://api.github.com/repos/pallets/flask/issues/comments/21494537",
            "html_url": "https://github.com/pallets/flask/issues/797#issuecomment-21494537",
            "issue_url": "https://api.github.com/repos/pallets/flask/issues/797",
            "id": 21494537,
            "node_id": "MDEyOklzc3VlQ29tbWVudDIxNDk0NTM3",
            "user": {
                "login": "untitaker",
                "id": 837573,
                "node_id": "MDQ6VXNlcjgzNzU3Mw==",
                "avatar_url": "https://avatars0.githubusercontent.com/u/837573?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/untitaker",
                "html_url": "https://github.com/untitaker",
                "followers_url": "https://api.github.com/users/untitaker/followers",
                "following_url": "https://api.github.com/users/untitaker/following{/other_user}",
                "gists_url": "https://api.github.com/users/untitaker/gists{/gist_id}",
                "starred_url": "https://api.github.com/users/untitaker/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/untitaker/subscriptions",
                "organizations_url": "https://api.github.com/users/untitaker/orgs",
                "repos_url": "https://api.github.com/users/untitaker/repos",
                "events_url": "https://api.github.com/users/untitaker/events{/privacy}",
                "received_events_url": "https://api.github.com/users/untitaker/received_events",
                "type": "User",
                "site_admin": false
            },
            "created_at": "2013-07-24T15:48:23Z",
            "updated_at": "2013-07-24T15:48:23Z",
            "author_association": "MEMBER",
            "body": "Apparently @mitsuhiko knows of this and therefore this is working as intended for now. The problem is finding a balance between intuitive behavior and memory efficiency. For more information read the IRC logs at http://dev.pocoo.org/irclogs/%23pocoo.2013-07-24.log\n"
        },
        {
            "url": "https://api.github.com/repos/pallets/flask/issues/comments/21494816",
            "html_url": "https://github.com/pallets/flask/issues/797#issuecomment-21494816",
            "issue_url": "https://api.github.com/repos/pallets/flask/issues/797",
            "id": 21494816,
            "node_id": "MDEyOklzc3VlQ29tbWVudDIxNDk0ODE2",
            "user": {
                "login": "mitsuhiko",
                "id": 7396,
                "node_id": "MDQ6VXNlcjczOTY=",
                "avatar_url": "https://avatars1.githubusercontent.com/u/7396?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/mitsuhiko",
                "html_url": "https://github.com/mitsuhiko",
                "followers_url": "https://api.github.com/users/mitsuhiko/followers",
                "following_url": "https://api.github.com/users/mitsuhiko/following{/other_user}",
                "gists_url": "https://api.github.com/users/mitsuhiko/gists{/gist_id}",
                "starred_url": "https://api.github.com/users/mitsuhiko/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/mitsuhiko/subscriptions",
                "organizations_url": "https://api.github.com/users/mitsuhiko/orgs",
                "repos_url": "https://api.github.com/users/mitsuhiko/repos",
                "events_url": "https://api.github.com/users/mitsuhiko/events{/privacy}",
                "received_events_url": "https://api.github.com/users/mitsuhiko/received_events",
                "type": "User",
                "site_admin": false
            },
            "created_at": "2013-07-24T15:52:03Z",
            "updated_at": "2013-07-24T15:52:03Z",
            "author_association": "MEMBER",
            "body": "That is an intentional change. The form parser is now using request.stream always which allows for greater flexibility and chunked request support.\n\nSome code is accessing request.data when it should not. Stop using request.data and all should be fine.\n\nNew code should use request.get_data and the old request data will be deprecated and removed in the future.\n"
        },
        {
            "url": "https://api.github.com/repos/pallets/flask/issues/comments/21511297",
            "html_url": "https://github.com/pallets/flask/issues/797#issuecomment-21511297",
            "issue_url": "https://api.github.com/repos/pallets/flask/issues/797",
            "id": 21511297,
            "node_id": "MDEyOklzc3VlQ29tbWVudDIxNTExMjk3",
            "user": {
                "login": "moodh",
                "id": 2162038,
                "node_id": "MDQ6VXNlcjIxNjIwMzg=",
                "avatar_url": "https://avatars0.githubusercontent.com/u/2162038?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/moodh",
                "html_url": "https://github.com/moodh",
                "followers_url": "https://api.github.com/users/moodh/followers",
                "following_url": "https://api.github.com/users/moodh/following{/other_user}",
                "gists_url": "https://api.github.com/users/moodh/gists{/gist_id}",
                "starred_url": "https://api.github.com/users/moodh/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/moodh/subscriptions",
                "organizations_url": "https://api.github.com/users/moodh/orgs",
                "repos_url": "https://api.github.com/users/moodh/repos",
                "events_url": "https://api.github.com/users/moodh/events{/privacy}",
                "received_events_url": "https://api.github.com/users/moodh/received_events",
                "type": "User",
                "site_admin": false
            },
            "created_at": "2013-07-24T19:51:49Z",
            "updated_at": "2013-07-24T19:51:49Z",
            "author_association": "NONE",
            "body": "So request.form should not be used anymore?\nShould I convert request.get_data() to a dict with libparse.qs_parse or something like that?\n"
        },
        {
            "url": "https://api.github.com/repos/pallets/flask/issues/comments/21511672",
            "html_url": "https://github.com/pallets/flask/issues/797#issuecomment-21511672",
            "issue_url": "https://api.github.com/repos/pallets/flask/issues/797",
            "id": 21511672,
            "node_id": "MDEyOklzc3VlQ29tbWVudDIxNTExNjcy",
            "user": {
                "login": "DasIch",
                "id": 182316,
                "node_id": "MDQ6VXNlcjE4MjMxNg==",
                "avatar_url": "https://avatars0.githubusercontent.com/u/182316?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/DasIch",
                "html_url": "https://github.com/DasIch",
                "followers_url": "https://api.github.com/users/DasIch/followers",
                "following_url": "https://api.github.com/users/DasIch/following{/other_user}",
                "gists_url": "https://api.github.com/users/DasIch/gists{/gist_id}",
                "starred_url": "https://api.github.com/users/DasIch/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/DasIch/subscriptions",
                "organizations_url": "https://api.github.com/users/DasIch/orgs",
                "repos_url": "https://api.github.com/users/DasIch/repos",
                "events_url": "https://api.github.com/users/DasIch/events{/privacy}",
                "received_events_url": "https://api.github.com/users/DasIch/received_events",
                "type": "User",
                "site_admin": false
            },
            "created_at": "2013-07-24T19:57:43Z",
            "updated_at": "2013-07-24T19:57:43Z",
            "author_association": "CONTRIBUTOR",
            "body": "`request.form` is fine but you shouldn't access `request.data` and `request.form`.\n"
        },
        {
            "url": "https://api.github.com/repos/pallets/flask/issues/comments/21512682",
            "html_url": "https://github.com/pallets/flask/issues/797#issuecomment-21512682",
            "issue_url": "https://api.github.com/repos/pallets/flask/issues/797",
            "id": 21512682,
            "node_id": "MDEyOklzc3VlQ29tbWVudDIxNTEyNjgy",
            "user": {
                "login": "moodh",
                "id": 2162038,
                "node_id": "MDQ6VXNlcjIxNjIwMzg=",
                "avatar_url": "https://avatars0.githubusercontent.com/u/2162038?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/moodh",
                "html_url": "https://github.com/moodh",
                "followers_url": "https://api.github.com/users/moodh/followers",
                "following_url": "https://api.github.com/users/moodh/following{/other_user}",
                "gists_url": "https://api.github.com/users/moodh/gists{/gist_id}",
                "starred_url": "https://api.github.com/users/moodh/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/moodh/subscriptions",
                "organizations_url": "https://api.github.com/users/moodh/orgs",
                "repos_url": "https://api.github.com/users/moodh/repos",
                "events_url": "https://api.github.com/users/moodh/events{/privacy}",
                "received_events_url": "https://api.github.com/users/moodh/received_events",
                "type": "User",
                "site_admin": false
            },
            "created_at": "2013-07-24T20:14:06Z",
            "updated_at": "2013-07-24T20:14:06Z",
            "author_association": "NONE",
            "body": "Oh, so my application broke because request.data was used in a debug log early. \":D\"\nWell, I guess removing it solves all problems I've had. :)\n"
        },
        {
            "url": "https://api.github.com/repos/pallets/flask/issues/comments/21631968",
            "html_url": "https://github.com/pallets/flask/issues/797#issuecomment-21631968",
            "issue_url": "https://api.github.com/repos/pallets/flask/issues/797",
            "id": 21631968,
            "node_id": "MDEyOklzc3VlQ29tbWVudDIxNjMxOTY4",
            "user": {
                "login": "mitsuhiko",
                "id": 7396,
                "node_id": "MDQ6VXNlcjczOTY=",
                "avatar_url": "https://avatars1.githubusercontent.com/u/7396?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/mitsuhiko",
                "html_url": "https://github.com/mitsuhiko",
                "followers_url": "https://api.github.com/users/mitsuhiko/followers",
                "following_url": "https://api.github.com/users/mitsuhiko/following{/other_user}",
                "gists_url": "https://api.github.com/users/mitsuhiko/gists{/gist_id}",
                "starred_url": "https://api.github.com/users/mitsuhiko/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/mitsuhiko/subscriptions",
                "organizations_url": "https://api.github.com/users/mitsuhiko/orgs",
                "repos_url": "https://api.github.com/users/mitsuhiko/repos",
                "events_url": "https://api.github.com/users/mitsuhiko/events{/privacy}",
                "received_events_url": "https://api.github.com/users/mitsuhiko/received_events",
                "type": "User",
                "site_admin": false
            },
            "created_at": "2013-07-26T16:29:59Z",
            "updated_at": "2013-07-26T16:29:59Z",
            "author_association": "MEMBER",
            "body": "Just for the record: The latest Werkzeug release changes the descriptor behavior so that doing this by accident will no longer cause pain and agony.\n"
        }
    ]
}